pydough/sqlglot/transform_bindings/mysql_transform_bindings.py

```diff
@@ -0,0 +1,694 @@
+                    this=stop,
+                    expression=sqlglot_expressions.Greatest(
+                        this=sqlglot_expressions.Add(
+                            this=expr_length, expression=start
+                        ),
+                        expressions=[sql_zero],
+                    ),
+                )
+
+        # start adjustment
+        if start_idx is not None and start_idx >= 0:
+            start = one_index_start
+        elif start_idx is not None and start_idx < 0:
+            start = sqlglot_expressions.Case(
+                ifs=[
+                    sqlglot_expressions.If(
+                        this=sqlglot_expressions.GT(
+                            this=expr_length,
+                            expression=sqlglot_expressions.Abs(this=start),
+                        ),
+                        true=start,
+                    )
+                ],
+                default=sql_one,
+            )
+
+        elif start_idx is None:
+            start = sql_one
+
+        result: SQLGlotExpression = sqlglot_expressions.Substring(
+            this=string_expr, start=start, length=length
+        )
+
+        return result
+
+    def convert_get_part(
+        self, args: list[SQLGlotExpression], types: list[PyDoughType]
+    ) -> SQLGlotExpression:
+        """
+        GETPART(str, delim, idx) ->
+            CASE
+                WHEN LENGTH(str) = 0 THEN NULL
+                WHEN LENGTH(delim) = 0 THEN
+                    CASE
+                        WHEN ABS(idx) = 1 THEN str
+                        ELSE NULL
+                    END
+                WHEN idx > 0 AND idx <= (LENGTH(str) - LENGTH(REPLACE(str, delim, '')))/LENGTH(delim) + 1
+                    THEN SUBSTRING_INDEX(SUBSTRING_INDEX(str, delim, idx), delim, -1)
+                WHEN idx < 0 AND -idx <= (LENGTH(str) - LENGTH(REPLACE(str, delim, '')))/LENGTH(delim) + 1
+                    THEN SUBSTRING_INDEX(SUBSTRING_INDEX(str, delim, idx), delim, 1)
+                WHEN idx = 0 THEN SUBSTRING_INDEX(SUBSTRING_INDEX(str, delim, 1), delim, -1)
+                ELSE NULL
+            END
+        """
+
+        assert len(args) == 3
+
+        string_expr, delimiter_expr, index_expr = args
+        literal_1: SQLGlotExpression = sqlglot_expressions.Literal.number(1)
+        literal_neg_1: SQLGlotExpression = sqlglot_expressions.Literal.number(-1)
+        literal_0: SQLGlotExpression = sqlglot_expressions.Literal.number(0)
+
+        # (LENGTH(str) - LENGTH(REPLACE(str, delim, '')))/LENGTH(delim) + 1
+        difference: SQLGlotExpression = sqlglot_expressions.Sub(
+            this=sqlglot_expressions.Length(this=string_expr),
+            expression=sqlglot_expressions.Length(
+                this=self.convert_replace([string_expr, delimiter_expr], types[:2])
+            ),
+        )
+
+        index_expr = apply_parens(index_expr)
+
+        delimiter_count: SQLGlotExpression = sqlglot_expressions.Div(
```

Comment:
should this be wrapped in a `FLOOR` to avoid decimal vs. truncate in div behavior?